controllers/admin/AdminImportController.php

```diff
@@ -3981,120 +5060,147 @@ public function supplyOrdersDetailsImport($offset = false, $limit = false, &$cro
-            }
-            if ($discount_rate < 0 || $discount_rate > 100) {
-                $this->errors[] = sprintf(
-                    $this->trans('Discount rate (%d) is not valid (at line %d). %s.', array(), 'Admin.Advparameters.Notification'),
-                    $discount_rate,
-                    $current_line + 1,
-                    $this->trans('Format: Between 0 and 100', array(), 'Admin.Advparameters.Notification')
-                );
-            }
-            if ($tax_rate < 0 || $tax_rate > 100) {
-                $this->errors[] = sprintf(
-                    $this->trans('Quantity Expected (%d) is not valid (at line %d).', array(), 'Admin.Advparameters.Notification'),
-                    $tax_rate,
-                    $current_line + 1,
+            if (false === ($supplierReference = ProductSupplier::getProductSupplierReference(
+                $idProduct,
+                $idProductAttribute,
+                $supplyOrder->id_supplier
+            ))) {
+                $this->errors[] = $this->trans('Product (%productId%/%attributeId%) is not available for this order (at line %line%).', array(
+                    '%productId%'   => $idProduct,
+                    '%attributeId%' => $idProductAttribute,
+                    '%line%'        => $currentLine + 1,
+                ), 'Admin.Advparameters.Notification');
+            }
+            if ($unitPriceTe < 0) {
+                $this->errors[] = $this->trans('Unit Price (tax excl.) (%price%) is not valid (at line %line%).', array(
+                    '%price%' => $unitPriceTe,
+                    '%line%'  => $currentLine + 1,
+                ), 'Admin.Advparameters.Notification');
+            }
+            if ($quantityExpected < 0) {
+                $this->errors[] = $this->trans('Quantity Expected (%qty%) is not valid (at line %line%).', array(
+                    '%qty%'  => $quantityExpected,
+                    '%line%' => $currentLine + 1,
+                ), 'Admin.Advparameters.Notification');
+            }
+            if ($discountRate < 0 || $discountRate > 100) {
+                $this->errors[] = $this->trans('Discount rate (%rate%) is not valid (at line %line%). %reason%.', array(
+                    '%rate%'   => $discountRate,
+                    '%line%'   => $currentLine + 1,
+                    '%reason%' => $this->trans('Format: Between 0 and 100', array(), 'Admin.Advparameters.Notification'),
+                ), 'Admin.Advparameters.Notification');
+            }
+            if ($taxRate < 0 || $taxRate > 100) {
+                $this->errors[] = $this->trans('Tax rate (%rate%) is not valid (at line %line%).', array(
+                    '%rate%' => $taxRate,
+                    '%line%' => $currentLine + 1,
                     $this->trans('Format: Between 0 and 100', array(), 'Admin.Advparameters.Notification')
```

Comment:
In both case, it seems `$this->trans('Format: Between 0 and 100', array(), 'Admin.Advparameters.Notification')` is never displayed